### ISSUE TITLE
Bug found when atom start up.

### ISSUE THREAD

**Issue by Psychelzh** [nesQuick/atom-wordcount#77]
[Enter steps to reproduce below:]

1. Start up atom.
2. ...

**Atom Version**: 1.12.6
**Electron Version**: 1.3.9
**System**: Unknown Windows Version
**Thrown From**: [wordcount](https://github.com/nesQuick/atom-wordcount) package, v2.10.1

### Stack Trace

Failed to activate the wordcount package

```
At Cannot read property 'getSelectedBufferRanges' of undefined

TypeError: Cannot read property 'getSelectedBufferRanges' of undefined
    at WordcountView.module.exports.WordcountView.getTexts (file:///C:/Users/liang/.atom/packages/wordcount/lib/wordcount-view.coffee:44:29)
    at WordcountView.getTexts (file:///C:/Users/liang/.atom/packages/wordcount/lib/wordcount-view.coffee:2:1)
    at WordcountView.module.exports.WordcountView.update_count (file:///C:/Users/liang/.atom/packages/wordcount/lib/wordcount-view.coffee:17:14)
    at file:///C:/Users/liang/.atom/packages/wordcount/lib/main.coffee:88:24
    at debounced (C:\Users\liang\.atom\packages\wordcount\node_modules\lodash\index.js:7737:27)
    at file:///C:/Users/liang/.atom/packages/wordcount/lib/main.coffee:101:7
    at Object.activate (file:///C:/Users/liang/.atom/packages/wordcount/lib/main.coffee:107:5)
    at Package.module.exports.Package.activateNow (C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package.js:184:19)
    at C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package.js:157:32
    at Package.module.exports.Package.measure (C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package.js:92:15)
    at C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package.js:150:26
    at Package.module.exports.Package.activate (C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package.js:147:34)
    at PackageManager.module.exports.PackageManager.activatePackage (C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package-manager.js:548:21)
    at C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package-manager.js:529:29
    at Config.module.exports.Config.transactAsync (C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\config.js:337:18)
    at PackageManager.module.exports.PackageManager.activatePackages (C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package-manager.js:524:19)
    at PackageManager.module.exports.PackageManager.activate (C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\package-manager.js:506:46)
    at C:\Users\liang\AppData\Local\atom\app-1.12.6\resources\app.asar\src\atom-environment.js:792:28
```

### Commands

```
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "markdown-preview"
    ],
    "telemetryConsent": "limited"
  },
  "wordcount": {
    "alwaysOn": true
  }
}
```

### Installed Packages

```coffee
# User
atom-html-preview, v0.1.22 (inactive)
highlight-selected, v0.11.2 (inactive)
language-latex, v1.0.0 (inactive)
language-matlab, v0.2.1 (inactive)
language-pfm, v0.82.0 (inactive)
language-vb, v0.1.0-pre1 (inactive)
markdown-preview-plus, v2.4.0 (inactive)
minimap, v4.25.7 (inactive)
minimap-git-diff, v4.3.1 (inactive)
minimap-highlight-selected, v4.4.0 (inactive)
split-diff, v1.1.0 (inactive)
wordcount, v2.10.1 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.45.0 (inactive)
base16-tomorrow-dark-theme, v1.3.0 (inactive)
base16-tomorrow-light-theme, v1.3.0 (inactive)
one-dark-ui, v1.6.2 (inactive)
one-light-ui, v1.6.2 (inactive)
one-dark-syntax, v1.5.0 (inactive)
one-light-syntax, v1.5.0 (inactive)
solarized-dark-syntax, v1.0.5 (inactive)
solarized-light-syntax, v1.0.5 (inactive)
about, v1.7.0 (inactive)
archive-view, v0.62.0 (inactive)
autocomplete-atom-api, v0.10.0 (inactive)
autocomplete-css, v0.13.1 (inactive)
autocomplete-html, v0.7.2 (inactive)
autocomplete-plus, v2.31.4 (inactive)
autocomplete-snippets, v1.11.0 (inactive)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (inactive)
background-tips, v0.26.1 (inactive)
bookmarks, v0.42.0 (inactive)
bracket-matcher, v0.82.2 (inactive)
command-palette, v0.39.0 (inactive)
deprecation-cop, v0.54.1 (inactive)
dev-live-reload, v0.47.0 (inactive)
encoding-selector, v0.22.0 (inactive)
exception-reporting, v0.40.0 (inactive)
find-and-replace, v0.202.2 (inactive)
fuzzy-finder, v1.4.0 (inactive)
git-diff, v1.1.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (inactive)
image-view, v0.60.0 (inactive)
incompatible-packages, v0.26.1 (inactive)
keybinding-resolver, v0.35.0 (inactive)
line-ending-selector, v0.5.0 (inactive)
link, v0.31.2 (inactive)
markdown-preview, v0.158.8 (inactive)
metrics, v1.1.2 (inactive)
notifications, v0.65.1 (inactive)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.1 (inactive)
settings-view, v0.243.1 (inactive)
snippets, v1.0.4 (inactive)
spell-check, v0.68.4 (inactive)
status-bar, v1.4.1 (inactive)
styleguide, v0.47.2 (inactive)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.0 (inactive)
timecop, v0.33.2 (inactive)
tree-view, v0.210.0 (inactive)
update-package-dependencies, v0.10.0 (inactive)
welcome, v0.35.1 (inactive)
whitespace, v0.35.0 (inactive)
wrap-guide, v0.38.2 (inactive)
language-c, v0.54.0 (inactive)
language-clojure, v0.22.1 (inactive)
language-coffee-script, v0.48.0 (inactive)
language-csharp, v0.13.0 (inactive)
language-css, v0.40.1 (inactive)
language-gfm, v0.88.0 (inactive)
language-git, v0.15.0 (inactive)
language-go, v0.43.0 (inactive)
language-html, v0.47.1 (inactive)
language-hyperlink, v0.16.1 (inactive)
language-java, v0.24.0 (inactive)
language-javascript, v0.122.0 (inactive)
language-json, v0.18.3 (inactive)
language-less, v0.29.6 (inactive)
language-make, v0.22.2 (inactive)
language-mustache, v0.13.0 (inactive)
language-objective-c, v0.15.1 (inactive)
language-perl, v0.37.0 (inactive)
language-php, v0.37.3 (inactive)
language-property-list, v0.8.0 (inactive)
language-python, v0.45.1 (inactive)
language-ruby, v0.70.2 (inactive)
language-ruby-on-rails, v0.25.1 (inactive)
language-sass, v0.57.0 (inactive)
language-shellscript, v0.23.0 (inactive)
language-source, v0.9.0 (inactive)
language-sql, v0.25.0 (inactive)
language-text, v0.7.1 (inactive)
language-todo, v0.29.1 (inactive)
language-toml, v0.18.1 (inactive)
language-xml, v0.34.12 (inactive)
language-yaml, v0.27.1 (inactive)

# Dev
No dev packages
```

**Comment by nesQuick** [comment]
@villermen do you have a thought on this one?

**Comment by villermen** [comment]
Apparently `atom.workspace.getActiveTextEditor()` can return `undefined` when the active item is not a TextEditor. This can probably easily be fixed by not calling `view.update_count` when there is no editor to supply it. I'll do some experimenting and get you a PR.

**Comment by villermen** [comment]
I haven't been able to reproduce the bug. However the proposed #78 should fix the issue, by simply not running update_count if it makes no sense.

**Comment by nesQuick** [comment]
Thanks @villermen looks good to me, I'm about to publish a new patch version

**Comment by nesQuick** [comment]
just published `v2.10.2` and hopefully fixed it. Big thanks again to @villermen 🎉 and please verify @Psychelzh 🐛 🔨

**Comment by Psychelzh** [comment]
Thanks! The new version `v2.10.2` solved the problem very well. @nesQuick
